lib/MetaCPAN/Query/Release.pm

```diff
@@ -1176,10 +1189,17 @@ sub find_download_url {
         );
     }
 
-    return +{
-        %{ $hit->{_source} },
-        %{ $hit->{inner_hits}{module}{hits}{hits}[0]{_source} }, @checksums,
-    };
+    my $source_name = delete $source->{name};
+    if ( !$module_filter ) {
+        $source->{release} = $source_name;
+    }
+
+    my $module
+        = $hit->{inner_hits}{module}
+        ? $hit->{inner_hits}{module}{hits}{hits}[0]{_source}
+        : {};
+
+    return +{ %$source, %$module, @checksums, };
```

Comment:
note that `$source` can be `undef`

Reply:
At this point, we have verified that there was a result in `$hit`. We are requesting specific `_source` fields. Even if none of those fields are available, `_source` will still exist unless Elasticsearch is entirely broken. `$source` can not be `undef`.